### ISSUE TITLE
Azure Java iot sdk never reconnecting back to the iot hub after connection loss.

### ISSUE THREAD

**Issue by milunko** [Azure/azure-iot-sdk-java#192]
We were using version 1.5.37 and were experiencing same issues as described in:
https://github.com/Azure/azure-iot-sdk-java/issues/148
Because of this we upgraded to 1.6.0 and currently we have a issue that our connection gets dropped after some time (1h of running or more) and SDK never reconnects back to IOT hub. 

Warn log from SDK produces:

```
2018-01-15 16:39:02,106 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_SUCCESS
2018-01-15 16:39:02,116 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_SUCCESS
2018-01-15 16:39:02,118 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_SUCCESS
2018-01-15 16:39:02,121 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_SUCCESS
2018-01-15 16:39:02,124 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_SUCCESS
2018-01-15 16:39:02,129 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_SUCCESS
2018-01-15 17:44:02,655 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 17:44:04,684 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 17:44:08,687 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 17:44:16,824 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 17:44:32,830 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 17:45:04,833 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 17:46:05,302 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 17:47:05,753 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 17:48:06,202 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 17:49:06,653 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 17:50:07,073 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 17:51:07,976 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 17:52:08,399 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 17:53:08,805 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 17:54:09,237 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 17:55:15,072 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 17:56:15,495 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 17:57:15,930 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 17:58:16,348 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 17:59:16,760 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 18:00:17,733 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 18:01:18,237 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 18:02:18,642 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 18:03:19,148 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 18:04:19,571 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 18:05:20,034 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 18:06:20,455 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 18:07:20,872 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 18:08:21,288 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 18:09:21,681 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 18:10:22,099 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 18:11:22,493 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 18:12:22,891 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 18:13:23,305 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 18:14:23,713 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 18:15:24,109 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 18:16:24,518 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 18:17:24,935 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 18:18:25,670 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-15 18:19:26,088 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
```
.... and so on....

So for the first one hour everything works fine, and then connection is never recovered.
If I close the connection manually and connect again to IOT hub same behaviour is repeated.

In the debug log information shown when connection is dropped:
```
2018-01-16 09:19:42,206 INFO c.m.a.s.i.d.t.a.AmqpsIotHubConnection [pool-11-thread-1] Proton reactor has been stopped, method name is closeAsync 
2018-01-16 09:19:42,210 INFO c.m.a.s.i.d.t.a.AmqpsIotHubConnection [pool-11-thread-1] Reactor is assigned to executor service, method name is openAsync 
2018-01-16 09:19:42,561 INFO c.m.a.s.i.d.t.a.AmqpsIotHubConnection [pool-11-thread-1] Starting to reconnect to IotHub, method name is onLinkRemoteClose 
2018-01-16 09:19:42,562 INFO c.m.a.s.i.d.t.a.AmqpsTransport [pool-11-thread-1] The messages in progress are buffered to be sent again due to a connection loss, method name is connectionLost 
2018-01-16 09:19:42,562 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-16 09:19:42,563 INFO c.m.a.s.i.d.t.a.AmqpsIotHubConnection [pool-11-thread-1] Lost connection to the server. Reconnection attempt 933, method name is startReconnect 
2018-01-16 09:20:42,564 INFO c.m.a.s.i.d.t.a.AmqpsIotHubConnection [pool-11-thread-1] Proton reactor has been stopped, method name is closeAsync 
2018-01-16 09:20:42,567 INFO c.m.a.s.i.d.t.a.AmqpsIotHubConnection [pool-11-thread-1] Reactor is assigned to executor service, method name is openAsync 
2018-01-16 09:20:42,943 INFO c.m.a.s.i.d.t.a.AmqpsIotHubConnection [pool-11-thread-1] Starting to reconnect to IotHub, method name is onLinkRemoteClose 
2018-01-16 09:20:42,945 INFO c.m.a.s.i.d.t.a.AmqpsTransport [pool-11-thread-1] The messages in progress are buffered to be sent again due to a connection loss, method name is connectionLost 
2018-01-16 09:20:42,945 WARN c.z.i.p.a.AzureIotHubBlock [pool-11-thread-1] Connection state changed to CONNECTION_DROP
2018-01-16 09:20:42,949 INFO c.m.a.s.i.d.t.a.AmqpsIotHubConnection [pool-11-thread-1] Lost connection to the server. Reconnection attempt 934, method name is startReconnect 
2018-01-16 09:20:57,191 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,197 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,207 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,218 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,228 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,237 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,247 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,257 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,267 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,277 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,287 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,297 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,307 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,317 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,328 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,337 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,347 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,357 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,367 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,377 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,387 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,397 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,408 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,417 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,427 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,437 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,447 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,457 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,467 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,477 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,488 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,497 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,508 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,517 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,527 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,537 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,547 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,557 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,567 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,584 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,588 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,597 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,608 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,617 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,627 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,637 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,647 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,657 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,667 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,677 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,687 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,697 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,707 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,718 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,727 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,737 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,747 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,757 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,768 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,777 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,787 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,797 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,807 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,817 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,827 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,837 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,847 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,857 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,867 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,877 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,887 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,897 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,907 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,917 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,927 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,937 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,948 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,957 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,967 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,977 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,988 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:57,997 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:58,007 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:58,017 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:58,028 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:58,037 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:58,047 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:58,057 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:58,068 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:58,078 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:58,087 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:58,097 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:58,107 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:58,117 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:58,127 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:58,137 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:58,147 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:58,157 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:58,168 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:58,177 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:58,187 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:58,197 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:58,207 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:58,217 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-2] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:58,227 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
2018-01-16 09:20:58,237 DEBUG c.m.a.s.i.d.t.a.AmqpsTransport [pool-12-thread-1] Get the callback function for the received message, method name is handleMessage 
```
...

and this pattern repeats with the same message in the log for ~1 minute.

Best,
Milos

**Comment by prmathur-microsoft** [comment]
Messages from your log are telling me that you lost connection to the server. Can you confirm you have network connectivity when this issue occurs ?

**Comment by milunko** [comment]
Yes, I have network connectivity and network is working fine. Stopping and starting the application temporarily fixes the issue but little time after same thing starts happening again (usually after 1h of idle time). 

This is not happening on one device but on multiple devices with v 1.6.0. As I mentioned with previous version this was not happening. We do have prod devices with 1.5.37 which are still running in the same network with no connection issues.

**Comment by milunko** [comment]
Any news?

**Comment by prmathur-microsoft** [comment]
We are able to reproduce this issue locally. We see several re connection attempts over AMQP often and yet connection isn't successful. Debug is in progress. I will keep you updated about the findings.

**Comment by syedhassaanahmed** [comment]
@prmathur-microsoft Any news on this?

**Comment by prmathur-microsoft** [comment]
We found couple of issues around using SAS token during reconnection. Can you confirm that you were using SAS Token ?

**Comment by milunko** [comment]
Hi, we are connecting to the IOT hub using the connection string provided.

**Comment by milunko** [comment]
Any news?

**Comment by timtay-microsoft** [comment]
This issue has been fixed as of [Device Client 1.9.0](https://github.com/Azure/azure-iot-sdk-java/releases/tag/2018-2-23)

I am closing this issue for now. Feel free to re-open if you still face issues
